UMA-Bots/Configuration-Change/src/utils.ts

```diff
@@ -0,0 +1,94 @@
+  "event SetPoolRebalanceRoute(uint256 indexed destinationChainId, address indexed l1Token, address indexed destinationToken)",
+  "event SetEnableDepositRoute(uint256 indexed originChainId, uint256 indexed destinationChainId, address indexed originToken, bool depositsEnabled)",
+  "event SpokePoolAdminFunctionTriggered(uint256 indexed chainId, bytes message)", // not exactly a configuration @Review
+];
+
+export const SPOKEPOOL_MONITORED_EVENTS = [
+  "event SetXDomainAdmin(address indexed newAdmin)",
+  "event SetHubPool(address indexed newHubPool)",
+  "event EnabledDepositRoute(address indexed originToken, uint256 indexed destinationChainId, bool enabled)",
+  "event SetDepositQuoteTimeBuffer(uint32 newBuffer)",
+];
+
+/*
+ * @desc Generates a dictionary from event name to the complete ABI of the event
+ * @dev For e.g. {"LivenessSet" : "event LivenessSet(uint256 newLiveness)"}
+ */
+export function generateDictNameToAbi(monitoredEvents: string[]) {
+  let res: Dictionary<string> = {};
+  for (let i = 0; i < monitoredEvents.length; i++) {
+    let name: string = monitoredEvents[i].split("(")[0].split(" ")[1];
+    res[name] = monitoredEvents[i];
+  }
+  return res;
+}
+
+/*
+ * @desc Returns the list of parameters for an event if provided the event ABI
+ * @dev For e.g. if input is ""event BondSet(address indexed newBondToken, uint256 newBondAmount)"
+ * @dev The output will be ["newBondToken","newBondAmount"]
+ */
+function eventToParamNames(eventAbi: string) {
+  eventAbi = eventAbi.substring(6).split(")")[0];
+  eventAbi = eventAbi.substring(6).split("(")[1];
+  let params: string[] = eventAbi.split(",");
+  let paramNames: string[] = [];
+  for (let i = 0; i < params.length; i++) {
+    let param: string[] = params[i].split(" ");
+    let paramName = param[param.length - 1];
+    paramNames.push(paramName);
+  }
+  return paramNames;
+}
+
+interface Dictionary<T> {
+  [Key: string]: T;
+}
+
+export function getEventMetadata(eventName: string, paramValues: any, eventNameToAbi: {}) {
+  let eventAbi = eventNameToAbi[eventName as keyof typeof eventNameToAbi];
+  return getEventMetadataFromAbi(eventAbi, paramValues);
+}
+
+/*
+ * @desc Returns the metadata to be returned in a finding
+ * @param eventAbi - ABI for the event
+ * @param paramValues - values for each of the parameters in the event
+ * @return the metadata dictionary with parameter names as keys and the passed values as values
+ */
+export function getEventMetadataFromAbi(eventAbi: string, paramValues: any[]) {
+  let paramNames: string[] = eventToParamNames(eventAbi);
+  let metadataDict: Dictionary<any> = {};
+  let argsDict: Dictionary<string> = {};
+  for (let i = 0; i < paramNames.length; i++) {
+    let paramName: string = paramNames[i];
+    argsDict[paramName] = paramValues[i].toString();
+  }
+  metadataDict["event"] = eventAbi.split("(")[0].split(" ")[1];
+  metadataDict["args"] = argsDict;
+
+  return metadataDict;
+}
```

Comment:
great work over here! but for readability and simplification, can you try just this function instead?

```suggestion
export const getMetadata = (args: { [key: string]: string }) => {
  const metadata: { [key: string]: string } = {};

  const allKeys: string[] = Object.keys(args);
  const keys: string[] = allKeys.slice(allKeys.length / 2);

  keys.forEach((key) => {
    metadata[key] = args[key].toString();
  });

  return metadata;
};

```

and in your agent file you can change `forEach` functions like this:

```
      hubPoolEventTxns.forEach((actualEventTxn) => {
        const args = getMetadata(actualEventTxn.args);
        const thisFindingMetadata = {
          event: actualEventTxn.name,
          args: args,
        };
        findings.push(getFindingInstance(true, thisFindingMetadata));
      });
```

and let me know if that worked

Reply:
It did, thanks. Much more efficient than before